front/src/App/model/talk.js

```diff
@@ -0,0 +1,21 @@
+export default class Talk {
+  constructor(id, name, description, meetingLink, speaker) {
+    this.id = id;
+    this.name = name;
+    this.description = description;
+    this.meetingLink = meetingLink;
+    this.speaker = speaker;
+  }
+
+  checkIsAssigned(slots) {
+    this.isAssigned = slots.some((slot) => slot.talk.id === this.id);
+  }
+
+  checkIsInqueue(queue) {
+    this.isInqueue = queue.some((talk) => talk === this);
+  }
+
+  checkIsToSchedule(openSpace) {
+    this.isToSchedule = openSpace.toSchedule.some((talk) => talk.id === this.id);
+  }
```

Comment:
En mi caso no estoy seguro de entender la intencion. Si se trata de responder 3 preguntas, por que se difiere la respuesta? Se la guarda como atributos de un objeto para futuro uso al parecer. Pero no tiene sentido response al toque en vez de guardarse las respuestas?

```javascript
isAssigned(slots) {
    return slots.some((slot) => slot.talk.id === this.id);
  }

  isInqueue(queue) {
    return queue.some((talk) => talk === this);

  isToSchedule(openSpace) {
    return openSpace.toSchedule.some((talk) => talk.id === this.id);
  }
```

La ventaja que le veo es que si mientras hiciste el check y usas la respuesta "paso algo" que cambio el estado interno del objeto, entonces no caerias en una inconsistencia. Porque los 3 checks se calculas _just in time_, segun el estado actual del objeto.

O sea, como diria Hernan, hay un problema de tiempo. Si el check es en dos pasos, se esta en riesgo de una inconsistencia. Eso no pasaria aca porque solo usas el id (que seguramente no va a cambiar), pero si pensamos esta practica en otro tipo de objetos, me parece un anti-pattern y lo evitaria.

Reply:
Entiendo que el objeto este en un estado invalido, o podría en algún momento estarlo, por hacer la verificación en un momento y "chequear" ese booleano en otro. Pero el tema es que no se puede hacer la verificación al momento de la creación, y si la verificación se quiere hacer al momento de su uso, tanto slots, como queue y openspace tienen que ir para todos los lugares que va la Talk. Siendo este el primer problema que se quería resolver, hacer que slots, queue y openspace no tengan que ser pasados a todo el arbol de componentes.